action.yml

```diff
@@ -1,27 +1,35 @@
 name: "Setup Scala with both Java and sbt"
 description: "Much longer description"
-#inputs:
-#  sbt-runner-version:
-#    description: "The runner version (The actual version is controlled via project/build.properties)"
-#    required: true
-#    default: 1.10.2
+inputs:
+  major-java-version:
+    description: 'Specify a major Java version (21, 11, etc) as an alternative to reading one from .tool-versions'
+    required: false
+    default: ''
 runs:
   using: "composite"
   steps:
     - name: Determine major Java version
       id: determine-java-version
       shell: bash
+      env:
+        MAJOR_JAVA_VERSION: ${{ inputs.major-java-version }}
       run: |
-        if [ ! -f .tool-versions ]; then
-          echo "::error title=Missing .tool-versions file::setup-scala requires an asdf-format .tool-versions file to establish the Java version for the build."
-          exit 1
+        if [ ${MAJOR_JAVA_VERSION:-} ]; then
```

Comment:
What is the difference between `${MAJOR_JAVA_VERSION:-}` and `${MAJOR_JAVA_VERSION}`?

Reply:
Ah, it looks like nothing useful! I was just trying to detect if the `MAJOR_JAVA_VERSION` was both set and non-blank, but just `${MAJOR_JAVA_VERSION}` seems do that:

```
% MAJOR_JAVA_VERSION="11"; if [ ${MAJOR_JAVA_VERSION} ]; then echo "var is set to '$MAJOR_JAVA_VERSION'"; else echo "var is blank"; fi
var is set to '11'
% unset MAJOR_JAVA_VERSION; if [ ${MAJOR_JAVA_VERSION} ]; then echo "var is set to '$MAJOR_JAVA_VERSION'"; else echo "var is blank"; fi
var is blank
% MAJOR_JAVA_VERSION=""; if [ ${MAJOR_JAVA_VERSION} ]; then echo "var is set to '$MAJOR_JAVA_VERSION'"; else echo "var is blank"; fi
var is blank
```

...I will update.

Reply:
Very good. Although I was hoping to learn about some more clever bash feature.